### ISSUE TITLE
Making dot-completion and object spread operator work together

### ISSUE THREAD

**Issue by unional** [Microsoft/vscode#17584]
- VSCode Version: 1.8.1

```ts
let x = { a: 1, b: 2 }
let y = {
  c: { d: 4 },
  ...x // <-- here
}
```

When you actually type the above code, the first `.` you type will trigger completion, showing `c`. If you press the second `.`, it will become `.c.` and should completion of `d`.

To avoid this, you have to press `[esc]` after you type the first `.`.

UPDATE: this occurs with `typescript@2.2.0-dev20161219`, but not in `typescript@2.2.0-dev20161206`.
Seems like a regression.

**Comment by joaomoreno** [comment]
Fix reverted 31997b32c354b0531f296379cec364b094ee362d, #18063

**Comment by unional** [comment]
Too bad. Need to re-open this issue for tracking?

**Comment by joaomoreno** [comment]
Oops, forgot to open, thanks @unional

**Comment by mjbvz** [comment]
#18380 addressed a similar bug in package names, but did not fix this scenario. Opened https://github.com/Microsoft/TypeScript/issues/13456 to track a requested change on their side that will help us fix this. Since that fix will target TS 2.2 at the earliest, I'll investigate another workaround here.
